common/src/id_generator.rs

```diff
@@ -0,0 +1,33 @@
+use crate::EntityId;
+use fxhash::FxHashMap;
+use hecs::Entity;
+use rand::rngs::SmallRng;
+use rand::{Rng, SeedableRng};
+
+pub struct IDGenerator {
```

Comment:
Since this struct contains the map from `EntityId` to `Entity`, it has become more than just an ID generator, so it likely deserves a different name, one that might help better define its purpose. Perhaps `EntityIdMap`?

However, I have second guesses about this kind of abstraction. Ideally, we should be able to spawn an entity, assign it an `EntityId`, and handle the `spawns`, `dirty_nodes`, and `graph_entities` fields all in a single method call, but the owner of that method would be `sim`, bringing us back to square one in terms of the borrow checker.

It pains me to suggest this, given my previous suggestion in Discord, but the best move for now might be to revert the creation of `id_generator`. I'm relatively unfamiliar with this part of the codebase and how to write good code for it, which is why my suggestions keep contradicting themselves.